library/x509_crl.c

```diff
@@ -294,7 +294,9 @@ int mbedtls_x509_crl_parse_der( mbedtls_x509_crl *chain,
     /*
      * Copy raw DER-encoded CRL
      */
-    if( buflen != 0 && ( ( p = mbedtls_calloc( 1, buflen ) ) == NULL ) )
+    if( buflen == 0 )
+        return( MBEDTLS_ERR_X509_INVALID_FORMAT );
+    else if( ( p = mbedtls_calloc( 1, buflen ) ) == NULL )
```

Comment:
It doesn't change the behavior, but I'd prefer not to put an `else` here, because it's misleading for readers: if `buflen==0`, the code below the `else` is no executed either. And since the result of the allocation is stored in a variable, there's no call for putting the assignment inside the `if`, so I'd prefer

```
if( buflen == 0 )
    return( MBEDTLS_ERR_X509_INVALID_FORMAT );
p = mbedtls_calloc( 1, buflen );
if( ( p == NULL )
    return( MBEDTLS_ERR_X509_ALLOC_FAILED );
```

Reply:
Ok, I think this is a really minor thing about coding style. I prefer the way I originally wrote it, but I changed the code to suit your preferred style in the latest commit.